src/app/config/groups/thunks.js

```diff
@@ -38,3 +37,84 @@ export const updateGroupRequest = (id, body) => dispatch => {
             console.error(error);
         });
 };
+
+export const deleteGroupRequest = id => dispatch => {
+    dispatch(actions.deleteGroupProgress());
+    teams
+        .deleteGroup(id)
+        .then(response => {
+            dispatch(actions.deleteGroupSuccess());
+            dispatch(push(url.resolve("../", true)));
+            notifications.add({ type: "positive", message: `Group "${response.name}" removed successfully`, autoDismiss: 5000 });
+        })
+        .catch(error => {
+            dispatch(actions.deleteGroupFailure());
+            if (error) {
+                notifications.add({ type: "warning", message: error?.message || error.status, autoDismiss: 5000 });
+            }
+            console.error(error);
+        });
+};
+
+export const fetchGroupsRequest = isNewSignIn => dispatch => {
+    dispatch(actions.loadGroupsProgress());
+    isNewSignIn
+        ? teams
+              .getGroups()
+              .then(response => {
+                  dispatch(actions.loadGroupsSuccess(response.groups));
+              })
+              .catch(error => {
+                  dispatch(actions.loadGroupsFailure());
+                  //TODO: map responses to user friendly by content designer
+                  if (error) {
+                      notifications.add({ type: "warning", message: error?.message || error.status, autoDismiss: 5000 });
+                  }
+                  console.error(error);
+              })
+        : teams
+              .getAll()
+              .then(response => {
+                  if (!response) return dispatch(actions.loadGroupsFailure());
+                  return dispatch(actions.loadGroupsSuccess(response));
```

Comment:
Just a question, Will the the remote api always return an array if the response is successful? If there are no groups available will it return an empty array?

Reply:
hi, I need to change that, I just copied this over.78 line is not needed